pyomo/contrib/fbbt/fbbt.py

```diff
@@ -729,10 +735,26 @@ def visiting_potential_leaf(self, node):
 
         if node.is_variable_type():
             lb, ub = self.bnds_dict[node]
-            if lb != -math.inf:
-                node.setlb(lb)
-            if ub != math.inf:
-                node.setub(ub)
+            if node.is_binary():
```

Comment:
Isn't this a special case of general integers?  Would it be difficult to change this that if the variable is discrete, then:
```
if node.is_binary() or node.is_integer():
    lb = max(lb, math.ceil(lb - self.binary_tol))
    ub = min(ub, math.floor(up + self.binary_tol))
    self.bnds_dict[node] = (lb, ub)
```
(The max / min is probably not necessary, but I include it for safety)

...and then it would make sense to rename `binary_tol` to `integer_tol`.

Reply:
@jsiirola Good suggestion. I just pushed this change. I like your approach a lot more than my if-else statements. Thanks.